src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/PrisonerIndexService.kt

```diff
@@ -71,15 +71,26 @@ class PrisonerIndexService(
   fun syncPrisoner(prisonerId: String): Prisoner? =
     nomisService.getOffender(prisonerId)?.let {
       reindex(offenderBooking = it)
-    } ?: run {
-      telemetryClient.trackEvent(
-        "POSOffenderNotFoundForIndexing",
-        mapOf("prisonerID" to prisonerId),
-        null,
-      )
-      log.error("POSOffenderNotFoundForIndexing {}", prisonerId)
-      return null
     }
+      ?: get(prisonerId)?.run {
+        delete(prisonerId)
+        telemetryClient.trackEvent(
+          "POSOffenderRemovedFromIndex",
+          mapOf("prisonerID" to prisonerId),
+          null,
+        )
+        log.info("POSOffenderRemovedFromIndex prisoner not in Nomis but was in index: {}", prisonerId)
+        return this
+      }
+      ?: run {
+        telemetryClient.trackEvent(
+          "POSOffenderNotFoundForIndexing",
+          mapOf("prisonerID" to prisonerId),
+          null,
+        )
+        log.error("POSOffenderNotFoundForIndexing {}", prisonerId)
+        return null
+      }
```

Comment:
This means _any_ sync operation will remove a nonexistent prisoner, not just a manual index update